lib/ansible/modules/system/xfs_quota.py

```diff
@@ -0,0 +1,393 @@
+  bhard:
+    description:
+      - Hard blocks quota limit.
+      - This argument supports human readable sizes.
+    type: str
+  bsoft:
+    description:
+      - Soft blocks quota limit.
+      - This argument supports human readable sizes.
+    type: str
+  ihard:
+    description:
+      - Hard inodes quota limit.
+    type: int
+  isoft:
+    description:
+      - Soft inodes quota limit.
+    type: int
+  rtbhard:
+    description:
+      - Hard realtime blocks quota limit.
+      - This argument supports human readable sizes.
+    type: str
+  rtbsoft:
+    description:
+      - Soft realtime blocks quota limit.
+      - This argument supports human readable sizes.
+    type: str
+  state:
+    description:
+      - Whether to apply the limits or remove them.
+      - When removing limit, they are set to 0, and not quite removed.
+    type: str
+    default: present
+    choices:
+      - present
+      - absent
+
+requirements:
+   - xfsprogs
+'''
+
+EXAMPLES = r'''
+- name: Set default project soft and hard limit on /opt of 1g
+  xfs_quota:
+    type: project
+    mountpoint: /opt
+    bsoft: 1g
+    bhard: 1g
+    state: present
+
+- name: Remove the default limits on /opt
+  xfs_quota:
+    type: project
+    mountpoint: /opt
+    state: absent
+
+- name: Set default soft user inode limits on /home of 1024 inodes and hard of 2048
+  xfs_quota:
+    type: user
+    mountpoint: /home
+    isoft: 1024
+    ihard: 2048
+
+'''
+
+RETURN = ''' # '''
+
+import grp
+import json
+import os
+import pwd
+import sys
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import human_to_bytes
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            bhard=dict(type='str'),
+            bsoft=dict(type='str'),
+            ihard=dict(type='int'),
+            isoft=dict(type='int'),
+            mountpoint=dict(type='str', required=True),
+            name=dict(type='str'),
+            rtbhard=dict(type='str'),
+            rtbsoft=dict(type='str'),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            type=dict(type='str', required=True, choices=['user', 'group', 'project'])
+        ),
+        supports_check_mode=True
+    )
+
+    quota_type = module.params['type']
+    name = module.params['name']
+    mountpoint = module.params['mountpoint']
+    bhard = module.params['bhard']
+    bsoft = module.params['bsoft']
+    ihard = module.params['ihard']
+    isoft = module.params['isoft']
+    rtbhard = module.params['rtbhard']
+    rtbsoft = module.params['rtbsoft']
+    state = module.params['state']
+
+    if bhard is not None:
+        bhard = human_to_bytes(bhard)
+
+    if bsoft is not None:
+        bsoft = human_to_bytes(bsoft)
+
+    if rtbhard is not None:
+        rtbhard = human_to_bytes(rtbhard)
+
+    if rtbsoft is not None:
+        rtbsoft = human_to_bytes(rtbsoft)
+
+    result = dict(
+        changed=False
```

Comment:
Last element has trailing comma.
```suggestion
        changed=False,
```